### PR TITLE
Add plugin and test

### PR DESCRIPTION
**Pull request by gassmoeller**
See changes entry. This is useful for melt models with pre-existing melt.

### CODE REVIEW

File: source/initial_composition/porosity.cc [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          // Use the initial composition, except for the porosity, to prevent
+          // infinite recursion
+          for (unsigned int i = 0; i < this->n_compositional_fields(); ++i)
+            if (i != porosity_index)
+              in.composition[0][i] = this->get_initial_composition_manager().initial_composition(position,i);
+            else
+              in.composition[0][i] = 0.0;
+
+          in.strain_rate[0] = SymmetricTensor<2,dim>();
+          in.cell= NULL;
```
Comment (tjhei) [inline source/initial_composition/porosity.cc]: `` = ``

File: tests/initial_porosity.prm [hunk 2]
```diff
@@ -0,0 +1,232 @@
+# Listing of Parameters
```
Comment (tjhei) [inline tests/initial_porosity.prm]: can you update the comment please? Sorry.

File: tests/initial_porosity.prm [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+  subsection Function
+    set Function expression = if (x*x+y*y < 2700000*2700000,500,0)
+  end
+end
+
+subsection Initial composition model
+  set Model name = porosity
+  subsection Function
+    set Function constants  = pi=3.1415926,a = 0.0, b = 2500000, c = 100000, d=1450000
```
Comment (tjhei) [inline tests/initial_porosity.prm]: unused, right? please remove.

File: source/initial_composition/porosity.cc [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  namespace InitialComposition
+  {
+    ASPECT_REGISTER_INITIAL_COMPOSITION_MODEL(Porosity,
+                                              "porosity",
+                                              "A class that implements initial conditions for the porosity field "
+                                              "by computing the equilibrium melt fraction for the given initial "
+                                              "condition and reference pressure profile. Note that this plugin only "
+                                              "works if there is a compositional field called 'porosity', and the "
+                                              "used material model implements the 'MeltFractionModel' interface. "
+                                              "All compositional fields except porosity are not changed by this plugin.")
```
Comment (tjhei) [inline source/initial_composition/porosity.cc]: Is this really true? You are setting all other fields to zero or not?

Comment (gassmoeller) [inline source/initial_composition/porosity.cc]: Yes my description is true. I return 0.0 for other fields, which means the manager will use the selected operation (add by default) to combine this initial condition with the others. If somebody set the operation to max for the current plugin (which makes no sense) then I would zero out other compositions. Should I add an assert for the operator? I am not even sure if that is possible at the moment (i.e. if a plugin can control how its return value is used).

**Comment by gassmoeller** [comment]
Addressed the other comments.

File: source/initial_composition/porosity.cc (hunk 4 above)
Comment (tjhei) [inline source/initial_composition/porosity.cc]: Maybe it is more appropriate to say that it returns 0 for other fields? (or in addition)?

**Comment by gassmoeller** [comment]
Better now?

**Pull request merged**